src/__tests__/matches.js

```diff
@@ -15,6 +15,14 @@ test('matchers accept regex', () => {
   expect(fuzzyMatches('ABC', node, /ABC/, normalizer)).toBe(true)
 })
 
+// https://stackoverflow.com/questions/1520800/why-does-a-regexp-with-global-flag-give-wrong-results
+test('matchers recreate regex to prevent global mistakes', () => {
```

Comment:
It's not really about re-creating regexps. That's an implementation detail which is evident by the fact that we no longer re-create but mutate.